pwiz_tools/Skyline/TestFunctional/FullScanGraphTest.cs

```diff
@@ -298,6 +298,39 @@ private void TestSpecialIonsAnnotations()
             //check that the special ion annotation shows in both library and full scan viewers
             TestAnnotations(new [] {"Reporter_Test+"});
             TestLibraryMatchAnnotations(new[] { "Reporter_Test+" });
+            Settings.Default.ShowSpecialIons = false;
+        }
+
+        private void TestIonMatchToleranceUnitSetting()
+        {
+            RunUI(() => {
+                SkylineWindow.ShowBIons(true);
+                SkylineWindow.ShowCIons(true);
+            });
+
+            SetZoom(false);
+            ClickChromatogram(33.11, 15.055, PaneKey.PRODUCTS);
+            WaitForGraphs();
+            Assert.AreEqual(Skyline.Program.SkylineOffscreen? 70 : 20, SkylineWindow.GraphFullScan.IonLabels.Count());

```

Comment:
Why are these numbers different on-screen vs. off-screen? This at least needs a comment explaining the issue.

Reply:
The labels are not drawn in off-screen mode because it happens during Paint event. In the off-screen mode this tests total number of peaks matching the ion type filter. I'll add a comment in the code.

Reply:
Another approach can be to implement a method that gets the ion labels that would be added if the graph were painted. But, looking for a good example, I just ran into another case where we simply look at the CurveList for TextObj items. And I can see some benefit to testing the text that actually ends up on the graph. So, I can live with this.